beater/beater.go

```diff
@@ -969,3 +976,45 @@ func (r *chanReloader) serve(ctx context.Context, reloader reload.Reloadable) er
 		}
 	}
 }
+
+// TODO: This is copying behavior from libbeat:
+// https://github.com/elastic/beats/blob/b9ced47dba8bb55faa3b2b834fd6529d3c4d0919/libbeat/cmd/instance/beat.go#L927-L950
+// Remove this when cluster_uuid no longer needs to be queried from ES.
+func queryClusterUUID(ctx context.Context, esClient elasticsearch.Client) error {
+	stateRegistry := monitoring.GetNamespace("state").GetRegistry()
+	outputES := "outputs.elasticsearch"
+	elasticsearchRegistry := stateRegistry.GetRegistry(outputES)
+	if elasticsearchRegistry == nil {
+		elasticsearchRegistry = stateRegistry.NewRegistry(outputES)
+	}
```

Comment:
Being overly defensive here for test purposes. This function is executed without the side-effects from libbeat, which was causing a nil-registry to be found in some tests.

Reply:
Perhaps we should create the registry in the test instead? Would be nice not to have test-related things in this code